service/chartconfig/chart_framework.go

```diff
@@ -35,6 +37,9 @@ func NewChartFramework(config ChartFrameworkConfig) (*framework.Framework, error
 	if config.K8sExtClient == nil {
 		return nil, microerror.Maskf(invalidConfigError, "config.K8sExtClient must not be empty")
 	}
+	if config.ApprClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.ApprClient must not be empty")
+	}
```

Comment:
IMO we should only validate where we actually use something. The app client is going to be validated at the end of the chain anyway. I think it is not necessary to validate in each layer. This is just unnecessary code.